packages/components/link/src/link.tsx

```diff
@@ -0,0 +1,143 @@
+import type { Theme } from '@emotion/react';
+import type { MessageDescriptor } from 'react-intl';
+import React from 'react';
+import { Link as ReactRouterLink } from 'react-router-dom';
+import { css, useTheme } from '@emotion/react';
+import { FormattedMessage } from 'react-intl';
+import { customProperties as vars } from '@commercetools-uikit/design-system';
+import { filterInvalidAttributes, warning } from '@commercetools-uikit/utils';
+
+type TExtendedTheme = Theme & {
+  [key: string]: string;
+};
+type TLinkShape = {
+  pathname: string;
+  search: string;
+  hash?: string;
+  state: unknown;
+};
+
+type TLinkProps = {
+  tone?: 'primary' | 'inverted';
+  // @TODO: deprecate?
+  hasUnderline: boolean;
+  isExternal: boolean;
+  intlMessage?: MessageDescriptor;
+  children: React.ReactNode;
+  to: string | TLinkShape;
+};
+
+const warnIfMissingContent = (props: TLinkProps) => {
+  const hasContent =
+    props.intlMessage || Boolean(React.Children.count(props.children));
+
+  warning(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`intlMessage\` is marked as required in \`Link\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+  warning(
+    hasContent,
+    [
+      'Warning: Failed prop type:',
+      `The prop \`children\` is marked as required in \`Link\``,
+      'but its value is `undefined`',
+    ].join(' ')
+  );
+};
+
+const defaultProps: Pick<TLinkProps, 'tone' | 'hasUnderline' | 'isExternal'> = {
+  tone: 'primary',
+  hasUnderline: true,
+  isExternal: false,
+};
+
+const getColorValue = (
+  tone: string = 'primary',
+  overwrittenVars: TExtendedTheme
+) => {
+  if (tone === 'primary') {
+    return overwrittenVars.colorPrimary;
+  }
+
+  return overwrittenVars.fontColorForTextWhenInverted;
+};
+const getActiveColorValue = (
+  tone: string = 'primary',
+  overwrittenVars: TExtendedTheme
+) => {
+  if (tone === 'primary') {
+    return overwrittenVars.colorPrimary25;
+  }
+
+  return overwrittenVars.fontColorForTextWhenInverted;
+};
+
+const getLinkStyles = (props: TLinkProps, theme: Theme) => {
+  const overwrittenVars = {
+    ...vars,
+    ...theme,
+  };
+
+  const color = getColorValue(props.tone, overwrittenVars);
+  const hoverColor = getActiveColorValue(props.tone, overwrittenVars);
+
+  return css`
+    font-family: inherit;
+    color: ${color};
+    font-size: ${overwrittenVars.fontSizeDefault};
+    text-decoration: ${props.hasUnderline ? 'underline' : 'none'};
+    &:hover,
+    &:focus,
+    &:active {
+      color: ${hoverColor};
+    }
+  `;
+};
+
+const Link = (props: TLinkProps) => {
+  const theme = useTheme();
+
+  const remainingProps = filterInvalidAttributes(props);
+  warnIfMissingContent(remainingProps as TLinkProps);
+
+  if (props.isExternal && typeof props.to === 'string') {
```

Comment:
There is probably a better way for type guarding. suggestions welcome

Reply:
Hmm in this case I would only keep the `if (props.isExternal)` condition.

If the `to` prop is a location object, we can construct the URL string on our own.

Reply:
good call.
I'll have a look at that.